### PR TITLE
BRS-629: Move edit button within accordions

### PR DESCRIPTION
**Pull request by molund**
### Jira Ticket:
BRS-629

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-629

### Description:
Implemented the layout shown in https://bcparksdigital.atlassian.net/browse/BUX-133 for desktop, and the mobile layout shown in BRS-629.

### CODE REVIEW

**Review by meredithom — COMMENTED** [review]
Couple small nits:

The edit button is working but there is a missing space between the icon and the text.
image

The spacing for the button also gets a bit wonky on the same screen break as the icons disappear, at 991px.
image

**Comment by molund** [comment]
@meredithom 
For the second screenshot you took, I was trying to keep the "Edit" button vertically aligned with the expand/collapse chevron.  But I can remove the code that does this and always keep it vertically centred if you think this looks wonky?

File: src/app/shared/components/accordion/accordion.component.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+$edit-col-width: 108px;
+$margin-overlap: 6px;
+$border-width: 2px;
+$border-radius: 4px;
+
+.accordion-header-row {
+  border: $border-width solid $primary-blue;
+  border-radius: $border-radius;
+
+  @media screen and (min-width: 768px) {
```
Comment (cameronpettit) [inline src/app/shared/components/accordion/accordion.component.scss:12]: This is a good use case for bootstrap's grid system, as is used by other components in this product. Breakpoints are unified so no hardcoding of pixels is required (typ).

File: src/app/shared/components/accordion/accordion.component.scss [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  border: $border-width solid $primary-blue;
+  border-radius: $border-radius;
+
+  // use a negative margin and solid background to overlap the
+  // accordion-header when the body is open, hiding the bottom
+  // border and rounded corner of the header
+  border-top: none;
+  border-top-left-radius: 0;
+  border-top-right-radius: 0;
+  margin-top: -$margin-overlap;
```
Comment (cameronpettit) [inline src/app/shared/components/accordion/accordion.component.scss:56]: Margins, padding, spacing, etc. can all also be handled by preexisting bootstrap classes to remain consistent with other components. We should rarely need to hardcode pixels. This `scss` works but likely will not play nice with future design changes (typ).

Comment (molund) [reply src/app/shared/components/accordion/accordion.component.scss:57]: I tried to use the `p-#` and `m-#` classes everywhere I could. But I had to do something a little bit hacky with negative margins and positive padding in this css to get the border working like the design without completely re-writing the accordions to use ng-bootstrap components. So I did end up setting a 6px negative margin and a 6px positive padding to explicitly hide the 4px rounded corner and the 2px border.

File: src/app/shared/components/accordion/accordion.component.scss (hunk 1 above)
Comment (molund) [reply src/app/shared/components/accordion/accordion.component.scss]: The appearance and behaviour of the accordion button is quite custom. I did the best I could to use built-in classes, but there is still a bit of pixel pushing here, unfortunately.

**Review by cameronpettit — APPROVED** [review]
Works as expected. Unable to find more ways to leverage more bootstrap classes in lieu of css.

**Review by meredithom — APPROVED** [review]

**Pull request merged**